src/engine/pieces/pawn.js

```diff
@@ -9,6 +9,26 @@ export default class Pawn extends Piece {
         this.name = PieceName.PAWN
     }
 
+    takeDiagonalPiece(board, color){
+        let currentSquare = board.findPiece(this);
+        let row = currentSquare.row;
+        let col = currentSquare.col;
+        let availableMoves = [];
+        let possibleDir = [-1, 1];
+        possibleDir.forEach(dir => {
+                let targetRow = row + color;
+                let targetCol = col + color * dir;
+                if (this.inBoundary(targetRow, targetCol)
+                    && (board.getPiece(Square.at(targetRow, targetCol)) !== undefined)
+                    && (board.getPiece(Square.at(targetRow, targetCol)).player !== this.player)
+                    && (board.getPiece(Square.at(targetRow, targetCol)).name !== PieceName.KING)
+                ) {
+                    availableMoves.push(Square.at(targetRow, targetCol))
+                }
+            }
+        )
+        return availableMoves;
```

Comment:
I think we can use the `createMovesArray` function for this, and then remove the squares that contain the king after. We can use the filter as before:
```
createMovesArray(...).filter(square => board.getPiece(...).name !== PieceName.KING)
```

We could even go one step further for this, and allow the createMovesArray function to take in an optional parameter called `filter` which would be a function similar to the one above